The 1.13 PR

zedwick opened the pull request:
Updated to 1.13. Added new blocks as-well as all the flattened and renamed blocks.

Everything seems to work. I have this running on a server with players and have had no issues yet. RestoreNature works, but is still very messy in oceans as it replaces everything with air. Am still working on that.

Changes version and built file name, also changes a lot of lines purely for formatting so I could read the code. (Sorry!)

zedwick committed:
It compiles! Updated to 1.13.

zedwick committed:
Removed mods blockId support

zedwick committed:
Add version to output file

zedwick committed:
removed netbeans junk

RoboMWM commented on pom.xml:
```diff
@@ -42,7 +42,7 @@
           </testResource>
         </testResources>
-        <finalName>${project.name}</finalName>
+        <finalName>${project.name}-${project.version}</finalName>
```
No.

RoboMWM commented on src/main/java/me/ryanhamshire/GriefPrevention/Claim.java:
```diff
@@ -145,7 +145,7 @@ public void removeSurfaceFluids(Claim exclusionClaim)
 					if(exclusionClaim != null && exclusionClaim.contains(block.getLocation(), true, false)) continue;
 					
-					if(block.getType() == Material.STATIONARY_LAVA || block.getType() == Material.WATER || block.getType() == Material.STATIONARY_WATER || block.getType() == Material.LAVA)
+					if(block.getType() == Material.LAVA || block.getType() == Material.WATER)
```
So both stationary and flowing variants are now considered one material?

FrankHeijden replied:
yes

RoboMWM commented on src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java:
```diff
@@ -3614,7 +3594,12 @@ public boolean pvpRulesApply(World world)
+                player.getStatistic(Statistic.PICKUP, Material.BIRCH_PLANKS) +
+                player.getStatistic(Statistic.PICKUP, Material.JUNGLE_PLANKS) +
+                player.getStatistic(Statistic.PICKUP, Material.ACACIA_PLANKS) +
+                player.getStatistic(Statistic.PICKUP, Material.DARK_OAK_PLANKS) > 0) return false;
```
Probably should include this in the PR description next time. Thanks though, been planning to "fix" this for some time now.

Jikoo replied:
Shouldn't it be LOG  variants, not PLANKS?

RoboMWM replied:
Ah yes, I believe that was the other mistake I made. Good catch, thanks.

RoboMWM reviewed (dismissed):
Other than the final jar name, looks good, and a very easy diff to peruse c:

 I guess full 1.13 compatibility can't be guaranteed with 1.12-compiled plugins despite what I was told otherwise. I'm going to be out a few weeks unfortunately (had 1.13 been released a month ago I may have had time), so I won't be able to test - but from what I've seen here so far it doesn't look like anything _should break_ so if anyone else wants to test, feel free to upload a jar here (might need to change extension to .zip to upload here, or can upload elsewhere).

RoboMWM committed:
Merge branch 'master' into 1.13

treydun commented on src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java:
```diff
@@ -706,15 +674,35 @@ else if(world.getEnvironment() == Environment.NORMAL)
         //default siege blocks
         this.config_siege_blocks = new ArrayList<Material>();
         this.config_siege_blocks.add(Material.DIRT);
+        this.config_siege_blocks.add(Material.GRASS_BLOCK);
```
I think this is supposed to be TALL_GRASS not GRASS_BLOCK. The previous version had LONG_GRASS which has been replaced by TALL_GRASS. I only noticed this because I made the same mistake at first editing this same exact line.

zedwick replied:
Here I changed GRASS to GRASS_BLOCK and LONG_GRASS to GRASS. What used to be DOUBLE_PLANT with data of 2 is now TALL_GRASS in 1.13.

I don't think changes here are needed (though adding TALL_GRASS seems sensible, it wasn't included originally and my intention was purely to update this for 1.13 without any other changes)

treydun replied:
You are absolutely correct. The TALL_GRASS being the archetype for GRASS, DEAD_BUSH and FERN was an oversight on my part. I apologize for the confusion. I did end up adding all 3 since DEAD_BUSH and FERN would no longer be covered by the blanket type check of 'TALL_GRASS' in versions prior to 1.13. I appreciate your hard work! It made it easier for me to finish my upgrade after I found it, and thank you for the reply.

RoboMWM replied:
My guess is that only material IDs were able to be put in this list, so if the ID for `TALL_GRASS` was present before, then the material(s) that it represented should be included as well, not just the first with data type 1.


Unless I'm not understanding what you said here.

treydun replied:
Yes, thats what I meant.. I mixed up TALL_GRASS and LONG_GRASS in my initial comment to zedwick. I did end up adding all 3 new material names in my fork after realizing my mistake. https://github.com/treydun/GriefPrevention/blob/1b894abe04c687f7c698fd7d642ce182b538c4b9/src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java#L700 I added them in a dedicated final enum class that I have been using to update all the plugins we use. https://github.com/treydun/GriefPrevention/blob/1b894abe04c687f7c698fd7d642ce182b538c4b9/src/main/java/com/buildatnight/utils/Untagged.java It is materials that are still lacking a grouping tag in spigot. I realize this is not ideal but I was tired of typing the same list of 16-17 colored material items out or 6 wood types of everything.

RoboMWM replied:
Sorry, was referring to @Zedwick's response, but as I re-read I might've been confused by your response as well. Is there any mapping to what the old/new Material enums are to the Minecraft blocks before I end up more confused... a quick search doesn't seem to reveal much about `LONG_GRASS` other than the `LongGrass` class, which idk if that's new or not.

treydun replied:
@RoboMWM No problem! The only mapping from old -> new materials is that spigot renamed old Material names to LEGACY_(Old material Name) That only helps plugins that are 1.12 and cant/wont be upgraded. Its strongly advised, especially once you put api-version: 1.13 in your plugin.yml to only use the new names. As far as Item names changing goes, on the vanilla end I use https://minecraft.gamepedia.com/1.13/Flattening That is the list of all name changes from mojangs perspective. Spigot/Bukkit also changed their material names to better match mojangs. The full list of current Material names in the bukkit enumset is here. https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html To help with grouping of items/blocks (because they all have individual names and dedicated classes) spigot also introduced the new Tag system. https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Tag.html# The items are tagged at the item registry level and then the Tag class in bukkit lets you pull enumsets of items tagged with whatever the enumsets name is. For example, Tag.SAPLINGS.getValues() is all the saplings Materials. You can also check if(Tag.SAPLINGS.isTagged(someitem)) Its not well documented yet and I havent figured out how to extend and tag my own items yet. Thats why I am using a helper enum currently

RoboMWM replied:
Oh I see, `GRASS` now refers to that specific variant of the one block high, passable stuff. Not the two block high stuff. I thought that was considered tall grass all this time... 

Anyways, its variants (dead bush and fern) should also be added then for a "true" 1.13 conversion.

treydun replied:
Yes, exactly. That was my confusion as well

RoboMWM committed:
revert unnecessary pom changes

Changes like these shouldn't be committed for PRs, as these
(especially version) will be modified on release.

RoboMWM committed:
Use log instead of plank

Originally my typo

RoboMWM committed:
add variants of grass to default siege blocks

RoboMWM committed:
Declare 1.13 api use in plugin.yml
